.github/workflows/e2e-tests.yml

```diff
@@ -21,15 +10,18 @@ on:
 
 jobs:
   build:
-    name: Run e2e tests
+    name: Run e2e tests (Node ${{ matrix.node-version }})
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: ['20.x', '21.x', '22.x']
```

Comment:
we should test on 20, 22, 23. don't really care about testing on unstable versions except for the one that's about to become stable